### PR TITLE
Swtich from task_info to libproc for system resource info

### PR DESCRIPTION
**Pull request by mlw**

### CODE REVIEW

File: Source/santad/main.mm [hunk 1]
```diff
@@ -56,13 +52,11 @@ static void SantaWatchdog(void *context) {
...
-  struct mach_task_basic_info info;
-  mach_msg_type_number_t task_info_count = MACH_TASK_BASIC_INFO_COUNT;
+  struct proc_taskinfo pti;
 
-  if (task_info(mach_task_self(), MACH_TASK_BASIC_INFO, (task_info_t)&info, &task_info_count) ==
-      KERN_SUCCESS) {
+  if (GetTaskInfo(&pti)) {
     // CPU
-    double total_time = (timeval_to_double(info.user_time) + timeval_to_double(info.system_time));
+    double total_time = (double)MachTimeToNanos(pti.pti_total_user + pti.pti_total_system);
```
Comment (kallsyms) [inline Source/santad/main.mm:59]: Do you need to divide by NSEC_PER_SEC here?

Comment (mlw) [reply Source/santad/main.mm]: Yep, good catch - divide by seconds on the next line so need the conversion.

File: Source/common/SystemResources.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+#import <Foundation/Foundation.h>
+#include <mach/mach_time.h>
+#include <sys/cdefs.h>
+#include <sys/proc_info.h>
+
+__BEGIN_DECLS
+
+// Get the mach timebase for the system
+mach_timebase_info_data_t GetTimebase();
```
[the remark this answers is not in the document]
Comment (mlw) [reply Source/common/SystemResources.h]: Done.

File: Source/common/SystemResources.h [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+// Convert mach absolute time to nanoseconds
+uint64_t MachTimeToNanos(uint64_t mach_time);
+
+// Get the result of proc_pidinfo with the PROC_PIDTASKINFO flavor
+BOOL GetTaskInfo(struct proc_taskinfo *pti_out);
```
[the remark this answers is not in the document]
Comment (mlw) [reply Source/common/SystemResources.h]: Sure, done. Required moving a couple files to ObjC++.

**Review by russellhancox — APPROVED** [review]

**Pull request merged**